tests/functional/test-mariadb-relay-drain.sh

```
@@ -0,0 +1,281 @@
+#!/bin/bash
+# Issue #106: MariaDB / SQL-stopped relay drain and false DeadMaster avoidance.
+# Works on MariaDB and MySQL topologies (SQL_THREAD stop + unapplied relay + failover).
+set -uo pipefail
+cd "$(dirname "$0")/../.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)